src/study_api.jl

```diff
@@ -0,0 +1,167 @@
+const PSRCLASSES_DEFAULTS = JSON.parsefile(joinpath(@__DIR__, "JSON", "psrclasses.defaults.json"))
+
+function create_element!(
+    data::Data,
+    name::String,
+)
+    if !haskey(PSRCLASSES_DEFAULTS, name)
+        error("Unknown PSR Class '$name'")
+    end
+
+    element = deepcopy(PSRCLASSES_DEFAULTS[name])
+
+    index = _insert_element!(data, name, element)
+
+    return index
+end
+
+function _insert_element!(
+    data::Data,
+    name::String,
+    element::Any,
+)
+    raw_data = _raw(data)
+
+    @assert raw_data isa Dict{String,<:Any}
+
+    if !haskey(raw_data, name)
+        error("PSR Class '$name' is not available for this study")
+    end
+
+    objects = raw_data[name]
+
+    @assert objects isa Vector{<:Any}
+
+    push!(objects, element)
+
+    return length(objects)
+end
+
+function _get_element(
+    data::Data,
+    name::String, # classe
+    index::Integer,
+)
+
+    # ~ Retrieves raw JSON-like dict, i.e. `Dict{String, Any}`.
+    # ~ `_raw(data)` is a safe interface for `data.raw`.
+    # ~ This dictionary was created by reading a JSON file.
+    raw_data = _raw(data)
+
+    @assert raw_data isa Dict{String,<:Any}
+
+    if !haskey(raw_data, name)
+        error("PSR Class '$name' is not available for this study")
+    end
+
+    # ~ Gathers a list containing all instances of the class referenced above.
+    objects = raw_data[name]
+
+    @assert objects isa Vector{<:Any}
+
+    if !(1 <= index <= length(objects))
+        error("Invalid index '$index' out of bounds [1, $(length(objects))]")
+    end
+
+    element = objects[index]
+
+    # ~ Stronger validation can be achieved by using JSONSchema.jl-like tools.
+    @assert element isa Dict{String,<:Any}
```

Comment:
Same here, you could assert the type by annotating the return type of this function and not by making a runtime `@assert`.
Validating that an element respects a JSON Schema is quite different than verifying that it is a `Dict{String, <:Any}`.

Reply:
In this case, the role played by the type assertion was similar to what is achieved by JSONSchema-validation against something like `{ "type" : "object" }`.

Reply:
Hmm got the idea, it could be some other thing like `{ "type" : 1}` element instead of another object. In either case, I think that `@assert` calls for checking types should be redesigned which seems to be the plan 😄